nemo/collections/tts/parts/preprocessing/features.py

```diff
@@ -0,0 +1,272 @@
+# Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import List, Optional, Union
+
+import librosa
+import numpy as np
+import torch
+
+from nemo.collections.asr.modules import AudioToMelSpectrogramPreprocessor
+from nemo.collections.tts.parts.utils.tts_dataset_utils import (
+    get_audio_paths_from_manifest,
+    get_feature_filename_from_manifest,
+)
+
+
+class Featurizer(ABC):
+    @abstractmethod
+    def setup(self, feature_dir: Path) -> None:
+        """
+        Setup machine for saving features to disk.
```

Comment:
If we switch to copying the audio directory structure instead of a single flat feature directory, then we have to create parent directories during the processing itself. So setup() is no longer needed.